wgpu/src/api/instance.rs

```diff
@@ -4,6 +4,23 @@ use crate::{dispatch::InstanceInterface, *};
 
 use std::{future::Future, sync::Arc};
 
+bitflags::bitflags! {
+    /// WGSL language extensions.
+    ///
+    /// WGSL spec.: <https://www.w3.org/TR/WGSL/#language-extensions-sec>
+    #[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, Hash)]
+    pub struct WGSLLanguageFeatures: u32 {
+        /// <https://www.w3.org/TR/WGSL/#language_extension-readonly_and_readwrite_storage_textures>
+        const ReadOnlyAndReadWriteStorageTextures = 1 << 0;
+        /// <https://www.w3.org/TR/WGSL/#language_extension-packed_4x8_integer_dot_product>
+        const Packed4x8IntegerDotProduct = 1 << 1;
+        /// <https://www.w3.org/TR/WGSL/#language_extension-unrestricted_pointer_parameters>
+        const UnrestrictedPointerParameters = 1 << 2;
+        /// <https://www.w3.org/TR/WGSL/#language_extension-pointer_composite_access>
+        const PointerCompositeAccess = 1 << 3;
+    }
+}
+
```

Comment:
question: Is there a reason this shouldn't be in `wgpu-types`? This would also have the benefit of not needing to add the `bitflags` dep. to `wgpu`.

Reply:
Not really, except that it is only used in wgpu.